contributors/design-proposals/core-metrics-pipeline.md

```diff
@@ -0,0 +1,295 @@
+  Logs *FilesystemUsage `json:"logs,omitempty"`  
+}  
+
+// CpuUsage holds statistics about the amount of cpu time consumed  
+type CpuUsage struct {  
+  // The time at which these stats were updated.  
+  Timestamp metav1.Time `json:"time"`  
+  // Average CPU usage rate over sample window (across all cores), in "cores".  
+  // The "core" unit represents nanoseconds of CPU time consumed per second.  
+  // For example, 5 nanocores means the process averaged 5 nanoseconds 
+  // of cpu time per second during the sample window.
+  // +optional  
+  UsageRateNanoCores *uint64 `json:"usageNanoCores,omitempty"`  
+  // Cumulative CPU usage (sum of all cores) since object creation.  
+  // +optional  
+  AggregateUsageCoreNanoSeconds *uint64 `json:"usageCoreNanoSeconds,omitempty"`  
+}  
+
+// MemoryUsage holds statistics about the quantity of memory consumed  
+type MemoryUsage struct {  
+  // The time at which these stats were updated.  
+  Timestamp metav1.Time `json:"time"`  
+  // The amount of working set memory. This includes recently accessed memory,  
+  // dirty memory, and kernel memory.  
+  // +optional  
+  UsageBytes *uint64 `json:"usageBytes,omitempty"`  
+}  
+
+// VolumeUsage holds statistics about the quantity of disk resources consumed for a volume  
+type VolumeUsage struct {  
+  // Embedded FilesystemUsage  
+  FilesystemUsage  
+  // Name is the name given to the Volume  
+  // +optional  
+  Name string `json:"name,omitempty"`  
+}  
+
+// FilesystemUsage holds statistics about the quantity of disk resources consumed  
+type FilesystemUsage struct {  
+  // The time at which these stats were updated.  
+  Timestamp metav1.Time `json:"time"`  
+  // The device on which resources are consumed  
+  Device string `json:"device"`  
+  // UsedBytes represents the disk space consumed on the device, in bytes.  
+  // +optional  
+  UsedBytes *uint64 `json:"usedBytes,omitempty"`  
+  // InodesUsed represents the inodes consumed on the device  
+  // +optional  
+  InodesUsed *uint64 `json:"inodesUsed,omitempty"`  
+}  
+```
+
+## Implementation Plan
+
+@dashpole will internally separate core metrics from summary metrics and make the kubelet use the core metrics.  
+@dashpole will create a separate endpoint TBD to publish this set of core metrics.  
+@dashpole will modify volume stats collection so that it relies on this code.   
+@dashpole will modify the structure of stats collection code to be "On-Demand".   
+
+Suggested, tentative future work, which may be covered by future proposals:  
+ - Obtain all runtime-specific information needed to collect metrics from the CRI.   
+ - Modify cAdvisor to be "stand alone", and run in a seperate binary from the kubelet.  It will consume the above metadata API, and provide the summary API.  
+ - The kubelet no longer provides the summary API, and starts, by default, cAdvisor stand-alone (which provides the summary API).  Include flag to disable running stand-alone cAdvisor.  
```

Comment:
nit: If we go the standalone cAdvisor root, I think it should be an optional component run as an add-on daemon set.